src/v/storage/ntp_config.h

```diff
@@ -47,6 +47,8 @@ class ntp_config {
         model::shadow_indexing_mode shadow_indexing_mode
           = model::shadow_indexing_mode::disabled;
 
+        std::optional<bool> read_replica;
```

Comment:
+1

Reply:
we have `bool read_replica` in topic properties, I don't want to change it cause otherwise I need to serialize `bool_class`. I think it's ok that read_replica's is `ntp_config` matches the type in `topic_properties`.